File: BackEnd/ApiFuturaAlicante/src/main/java/es/impulsalicante/ApiFuturaAlicante/dto/PrevisualizacionDTO.java
package es.impulsalicante.ApiFuturaAlicante.dto;

import es.impulsalicante.ApiFuturaAlicante.models.Imagen;

import java.util.List;

public class PrevisualizacionDTO {
    private ResumenMemoriaDTO resumen;
    private List<Imagen> imagenes;

    public List<Imagen> getImagenes() {
        return imagenes;
    }

    public ResumenMemoriaDTO getResumen() {
        return resumen;
    }

    public void setImagenes(List<Imagen> imagenes) {
        this.imagenes = imagenes;
    }

    public void setResumen(ResumenMemoriaDTO resumen) {
        this.resumen = resumen;
    }
}

File: FrontEnd/ng-task-19-2/src/app/modelos/previsualizacion.dto.ts
import { ResumenMemoriaDTO } from './resumen-memoria.dto';

interface Imagen {
  id: number;
  nombre: string;
  url: string;
  categoria?: string;
  seleccionada?: boolean;
}

export interface ImagePlaceholder {
  id: string
  name: string
  section: string
  description: string
  width: number
  height: number
  uploadedImage?: string
}

export interface UploadedImage {
  id: string
  name: string
  url: string
  file: File
}

export interface PdfGenerationRequest {
  year: number
  imageMapping: { [key: string]: string }
}


export interface PrevisualizacionDTO {
  resumen: ResumenMemoriaDTO;
  imagenes: Imagen[];
}
